File: cap07-maven/Carros/src/main/java/br/com/livro/domain/CarroService.java
package br.com.livro.domain;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

public class CarroService {
	private CarroDAO db = new CarroDAO();

	// Lista todos os carros do banco de dados
	public List<Carro> getCarros() {
		try {
			List<Carro> carros = db.getCarros();
			return carros;
		} catch (SQLException e) {
			e.printStackTrace();
			return new ArrayList<Carro>();
		}
	}

	// Busca um carro pelo id
	public Carro getCarro(Long id) {
		try {
			return db.getCarroById(id);
		} catch (SQLException e) {
			return null;
		}
	}

	// Deleta o carro pelo id
	public boolean delete(Long id) {
		try {
			return db.delete(id);
		} catch (SQLException e) {
			return false;
		}
	}

	// Salva ou atualiza o carro
	public boolean save(Carro carro) {
		try {
			db.save(carro);
			return true;
		} catch (SQLException e) {
			return false;
		}
	}

	// Busca o carro pelo nome
	public List<Carro> findByName(String name) {
		try {
			return db.findByName(name);
		} catch (SQLException e) {
			return null;
		}
	}

	public List<Carro> findByTipo(String tipo) {
		try {
			return db.findByTipo(tipo);
		} catch (SQLException e) {
			return null;
		}
	}
}

File: cap18-client-web-AngularJS/arquivos_angular/resources/js/services/carroService.js
/**
 * Arquivo que mantém a lista o estado da lista de carros. 
 */
(function() {

	angular.module('carroService', []);
	
	function carro() {
		// lista de carros
		var carros = [];

		function setCarros(carros) {
			this.carros = carros;
		}
		
		function getCarros() {
			return this.carros;
		}
		
		return {
			carros: carros,
			set : setCarros,
			get : getCarros
		}
	}
	
	angular.module('carroService').factory('carroService', carro);
}());